tuf/fetcher.py

```diff
@@ -0,0 +1,201 @@
+"""
+<Program Name>
+  fetcher.py
+
+<Author>
+  Teodora Sechkova <tsechkova@vmware.com>
+
+<Started>
+  December 14, 2020
+
+<Copyright>
+  See LICENSE-MIT OR LICENSE for licensing information.
+
+<Purpose>
+  Provides an interface for network IO abstraction.
+"""
+
+import abc
+import requests
+import six
+import logging
+import time
+
+import urllib3.exceptions
+import tuf.exceptions
+
+logger = logging.getLogger(__name__)
+
+
+class FetcherInterface():
+  """
+  <Purpose>
+  Defines an interface for abstract network download which can be implemented
+  for a variety of network libraries and configurations.
+  """
+  __metaclass__ = abc.ABCMeta
+
+
+  @abc.abstractmethod
+  def fetch(self, url, required_length):
+    """
+    <Purpose>
+      Fetches the contents of HTTP/HTTPS url from a remote server up to
+      required_length and returns a bytes iterator.
+
+    <Arguments>
+      url:
+        A URL string that represents the location of the file.
+
+      required_length:
+        An integer value representing the length of the file in bytes.
+
+    <Exceptions>
+      tuf.exceptions.SlowRetrievalError, if a timeout occurs while receiving
+      data from a server
+
+    <Returns>
+      A bytes iterator
+    """
+    raise NotImplementedError # pragma: no cover
+
+
+
+class FetcherSettings():
```

Comment:
Can you explain the construct? I can see it should work but what is the advantage of a separate FetcherSettings and RequestsFetcher?

Reply:
The idea was to separate `FetcherSettings` so that other `FetcherInterface` implementations can reuse them (by inheriting) or replace them with their own. But given that we cannot remove them from `tuf.settings` just like that, I dropped this change too.